CredentialProvider.Microsoft/CredentialProviders/Vsts/MSAL/MsalTokenProvider.cs

```diff
@@ -166,29 +176,45 @@ public async Task<IMsalToken> AcquireTokenWithWindowsIntegratedAuth(Cancellation
 
                 throw;
             }
-            finally
-            {
-                var helper = await GetMsalCacheHelperAsync();
-                helper?.UnregisterCache(publicClient.UserTokenCache);
-            }
        }
 
-        private async Task<IPublicClientApplication> GetPCAAsync(bool useLocalHost = false)
+        private async Task<IPublicClientApplication> GetPCAAsync(bool withBroker, bool useLocalHost)
         {
             var helper = await GetMsalCacheHelperAsync().ConfigureAwait(false);
 
             var publicClientBuilder = PublicClientApplicationBuilder.Create(this.clientId)
-                .WithAuthority(this.authority);
+                .WithAuthority(this.authority)
+                .WithLogging(
+                    (LogLevel level, string message, bool containsPii) => {
+                        if (containsPii)
+                        {
+                            this.Logger.Verbose($"MSAL Log ({level}): [REDACTED FOR PII]");
```

Comment:
You don't need to do this, you can just set "use PII" to false - this is an optional param of this method. 

Also, have a look at the overload with `IIDentityLogger`, which is easier to control.